### ISSUE TITLE
[Question] Server side Ad insertion and duration of content

### ISSUE THREAD

**Issue by nimdokai** [google/ExoPlayer#3523]
**Description**
We have ads that are inserted on server side (SSAI), which cause that we have only one stream for content(with ads).
Right now, if we've got a PREROLL with duration of 60 seconds and movie with duration of 10 min and 30 sec, we display (exo_position / exo_duration) time 1:01 / 11:30 - which basically means that we are in 1st second of movie and we've watched ad already.
Also on DefaultTimeBar(exo_progress) our position is appropriate to exo_position 1:01.

**Questions**
1. We'd like to not include ad time in displayed exo_position / exo_duration time,
(in this particular example would be (exo_position / exo_duration) 0:01 / 10:00) is it possible?

2. Also we'd like to ads duration does not effect DefaultTimeBar(exo_progress),
(in this particular example it would be appropriate  to exo_position 0:01) is it possible?

**Comment by andrewlewis** [comment]
One way to do this would be to subclass `DefaultTimeBar` and override `setPosition` etc. to take into account the preroll ad.

**Comment by nimdokai** [comment]
Thanks for respond. 
Changing `setPosition` or `SetDuration` affect my custom `TimeBar`, but it does not help with `exo_position` and `exo_duration Views` as those items are updated in `PlaybackControlView.updateProgress()`, does it mean it is also necessary to override `PlaybackControlView`?

**Comment by andrewlewis** [comment]
Replacing `PlaybackControlView` would work. Another option that might require less code would be to wrap the `SimpleExoPlayer` instance you pass to `setPlayer`, delegating all methods to the real player but fixing up the position/duration to the values you need.

**Comment by nimdokai** [comment]
Wraping `SimpleExoPlayer` had some issues with `TimeBar`, scrolling cause 'jump' to correct position. 
I end up with replacing `PlaybackControlView` - I have added some code in `updateProgress()`, in `seekTo(int windowIndex, long positionMs)` I have changed `dispatchSeekTo()` argument `positionMs` to inculde Ads Time also I've created additional `Interface` to get information about ads and it works fine.
Thanks for help.
